docs/research/Orange_Pi_flasher.md

```diff
@@ -0,0 +1,79 @@
+# Orange Pi flasher
+
+## Contents
+
+<!-- TOC -->
+
+- [Orange Pi flasher](#orange-pi-flasher)
+    - [Contents](#contents)
+    - [Installing armbian on Orange Pi](#installing-armbian-on-orange-pi)
+    - [Installing flashrom on Orange Pi](#installing-flashrom-on-rpi)
+    - [Connection](#connection)
+    - [Flashing](#flashing)
+
+<!-- /TOC -->
+
+## Installing armbian on Orange Pi
+
+We need to prepare a SD card with armibian image on it first.
+Downlaod The image from
+[here](https://dl.armbian.com/orangepizero/Debian_jessie_dev.7z)
+and extract it to your SD card.
+
+## Installing flashrom on Orange Pi
+
+Put the SD card into Orange Pi and boot it. Then install flashrom using the
+following commands:
+```
+git clone https://github.com/flashrom/flashrom.git
+cd flashrom
+make CONFIG_ENABLE_LIBPCI_PROGRAMMERS=no install
+```
+
+We also need to enable SPI:
+```
+echo "overlays=spi-spidev" >> /boot/armbianEnv.txt
+echo "param_spidev_spi_bus=1" >> /boot/armbianEnv.txt
+reboot
+```
+> Important! Edit the armbianEnv.txt file only once.
+
+## Connection
+
+Orange Pi pinout:
+
+![Orange Pi Pinout](https://i1.wp.com/oshlab.com/wp-content/uploads/2016/11/Orange-Pi-Zero-Pinout-banner2.jpg)
+
+
+ Orange Pi pins | APU2 pin J6
+:--------------:|:----------:
+ GND            | 2
+ SPI1_CS        | 3
+ SPI1_CLK       | 4
+ SPI1_MISO      | 5
+ SPI1_MOSI      | 6
+ 
+ 
+## Flashing
+
+After Orange Pi reboot type following command:
```

Comment:
Is this the full procedure? What about putting the APU to S5 power state before flashing?